core/tests/performance_framework/selenium_driver.py

```diff
@@ -0,0 +1,237 @@
+            raise ValueError(error_msg)
+
+        self.server = self.driver = self.proxy = None
+
+    def get_har_dict(self, page_url):
+        """Retrieves an HTTP Archive (HAR) dict for a given page URL.
+
+        The HAR dict is retrieved using the proxy server by recording the
+        communication between the client and server.
+        """
+        with Xvfb() as _xvfb:
+            self._setup_proxy()
+            self._setup_driver(use_proxy=True)
+
+            self.proxy.new_har(page_url, options={'captureHeaders': True})
+
+            self.driver.get(page_url)
+
+            # Wait for the complete page to load, otherwise XHR requests made
+            # after page load will not be recorded.
+            time.sleep(WAIT_DURATION)
+
+        har_dict = self.proxy.har
+
+        self._stop_proxy()
+        self._stop_driver()
+
+        return har_dict
+
+    def get_har_dict_cached_state(self, page_url):
+        """Retrieve HTTP Archive (HAR) or page session stats for a page URL
+        while simulating a cached state i.e, a return user.
+
+        To get it we require the use of a proxy server as we need to record the
+        communication between server and client.
+        """
+        with Xvfb() as _xvfb:
+            self._setup_proxy()
+            self._setup_driver(use_proxy=True)
+
+            # Fetch the page once. This leads to caching of various resources.
+            self.driver.get(page_url)
+
+            # Start recording har data for the next page fetch.
+            self.proxy.new_har(page_url, options={'captureHeaders': True})
+            self.driver.get(page_url)
+
+            # Wait for the complete page to load, otherwise XHR requests made
+            # after page load will not be recorded.
+            time.sleep(WAIT_DURATION)
+
+        har_dict = self.proxy.har
+
+        self._stop_proxy()
+        self._stop_driver()
+
+        return har_dict
+
+    def get_page_session_timings(self, page_url):
+        """Retrieve page load timings for a page URL."""
+        with Xvfb() as _xvfb:
+            self._setup_driver(use_proxy=False)
+
+            self.driver.get(page_url)
+            # Wait for the complete page to load, otherwise XHR requests made
+            # after page load will not be recorded.
+            time.sleep(WAIT_DURATION)
+
+            page_session_timings = (
+                self.driver.execute_script("return window.performance"))
+
+
+        self._stop_driver()
+
+        return page_session_timings
+
+    def get_page_session_timings_cached_state(self, page_url):
+        """Retrieve page load timings for a page URL while simulating a
+        cached state i.e, a return user.
+        """
+        with Xvfb() as _xvfb:
+            self._setup_driver(use_proxy=False)
+
+            # Fetch the page once. This leads to caching of various resources.
+            self.driver.get(page_url)
+            # Wait for the complete page to load, otherwise XHR requests made
```

Comment:
This comment doesn't need to be repeated throughout the class. Consider implementing a private method that includes this comment and the ``time.sleep(WAIT_DURATION)`` code and call it here and elsewhere. The name of the method should be enough to explain the reason to wait here, e.g. ``_wait_until_page_load_is_finished``.